crates/payload/builder/src/service.rs

```diff
@@ -271,17 +288,33 @@ where
     }
 }
 
-impl<Gen> Future for PayloadBuilderService<Gen>
+impl<Gen, St, Tasks> Future for PayloadBuilderService<Gen, St, Tasks>
 where
-    Gen: PayloadJobGenerator + Unpin + 'static,
-    <Gen as PayloadJobGenerator>::Job: Unpin + 'static,
+    Tasks: TaskSpawner +  Unpin + 'static,
+    Gen: PayloadJobGenerator<Tasks> + Unpin + Clone + 'static,
+    <Gen as PayloadJobGenerator<Tasks>>::Job: Unpin + 'static,
+    St: CanonStateSubscriptions + Send + Unpin + 'static,
 {
     type Output = ();
 
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         let this = self.get_mut();
-
         loop {
+            let mut canon_events = this.chain_events.subscribe_to_canonical_state();
```

Comment:
oh actually, we can simplify all of this a lot.

sorry for the confusion.

we can just given the payloadbuilder service an instance of

https://github.com/paradigmxyz/reth/blob/b1d0dc41c2381951c0153a20dc1381b2434bca2a/crates/storage/provider/src/traits/chain.rs#L39-L39

obtained in the `spawn_payload_service` function, then we don't need any of the additional generics and can just poll the stream here